### PR TITLE
PCX-2789: Charge Googlepay throug Braintree provider

### PR DESCRIPTION
**Pull request by siebeprojects**
## WHAT

- https://optile.atlassian.net/browse/PCX-2789
- New GooglePay - Braintree payment service handling payments using the 3rd party Braintree SDK.

## WHY

As an end-customer I want to make a payment with GooglePay so that I can use my already saved payment methods (on GooglePay) to check out.

### CODE REVIEW

File: paymentservice/googlepay-braintree/src/main/java/com/payoneer/checkout/payment/googlepaybraintree/GooglePayBraintreePaymentService.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+import android.text.TextUtils;
+import android.util.Log;
+
+/**
+ * Create a new GooglePayBraintreePaymentService, this service handles payment requests
+ * for GooglePay networks routed through Braintree provider.
+ */
+public class GooglePayBraintreePaymentService extends PaymentService {
+
+    final static String TAG = "GooglePayBraintree";
```
Comment (Fbada006) [inline paymentservice/googlepay-braintree/src/main/java/com/payoneer/checkout/payment/googlepaybraintree/GooglePayBraintreePaymentService.java:46]: This is good

**Review by Fbada006 — APPROVED** [review]

File: checkout/src/main/java/com/payoneer/checkout/ui/screen/payment/ProcessPaymentViewModel.java [hunk 2]
```diff
@@ -135,9 +145,12 @@ private void setCloseWithCheckoutResult(final CheckoutResult checkoutResult) {
         closeWithCheckoutResult.setValue(new ContentEvent<>(checkoutResult));
     }
 
-    private void setShowProcessPaymentProgress(final Boolean visible) {
+    private void setShowProcessPaymentProgress(final boolean visible, final boolean finalizing) {
+        String header = finalizing ? Localization.translate(CHARGE_TITLE) : null;
```
[the remark this answers is not in the document]
Comment (siebeprojects) [reply checkout/src/main/java/com/payoneer/checkout/ui/screen/payment/ProcessPaymentViewModel.java:149]: It is used as a flag for the PaymentService to indicate that the payment will be finalized and preferably should not be interrupted by the end-user. The loading/progress animation changes when this flag is set to true.

File: paymentservice/googlepay-braintree/src/main/java/com/payoneer/checkout/payment/googlepaybraintree/GooglePayBraintreeFragment.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    public static GooglePayBraintreeFragment newInstance(final Bundle arguments) {
+        GooglePayBraintreeFragment fragment = new GooglePayBraintreeFragment();
+        fragment.setArguments(arguments);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
```
[the remark this answers is not in the document]
Comment (siebeprojects) [reply paymentservice/googlepay-braintree/src/main/java/com/payoneer/checkout/payment/googlepaybraintree/GooglePayBraintreeFragment.java]: This has been changed now.

File: checkout/src/main/java/com/payoneer/checkout/ui/screen/shared/ProgressSettings.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * See the LICENSE file for more information.
+ */
+
+package com.payoneer.checkout.ui.screen.shared;
+
+public final class ProgressSettings {
+    public final boolean visible;
+    public final String header;
+    public final String info;
+
```
[the remark this answers is not in the document]
Comment (siebeprojects) [reply checkout/src/main/java/com/payoneer/checkout/ui/screen/shared/ProgressSettings.java:15]: The header and info texts are not always shown for the progress view.
This class will allow customizing when to show / hide the header and info texts.

**Review by Fbada006 — APPROVED** [review]

**Review by caiomello — APPROVED** [review]

**Review by caiomello — APPROVED** [review]

**Review by Fbada006 — APPROVED** [review]

**Pull request merged**